x-pack/plugins/security_solution/public/common/lib/apm/use_track_http_request.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import { useStartTransaction } from './use_start_transaction';
+
+interface UseTrackHttpRequestOptions {
+  name: string;
+  spanName?: string;
+}
+
+export type RequestResult = 'success' | 'error' | 'aborted' | 'malformed';
+
+export const useTrackHttpRequest = () => {
+  const { startTransaction } = useStartTransaction();
+
+  const startTracking = useCallback(
+    ({ name, spanName = 'fetch' }: UseTrackHttpRequestOptions) => {
+      // Create the transaction, the managed flag is turned off to prevent it from being polluted by non-related automatic spans.
+      // The managed flag can be turned on to investigate high latency requests in APM.
+      // However, note that by enabling the managed flag, the transaction trace may be distorted by other requests information.
+      const transaction = startTransaction({
+        name,
+        type: 'http-request',
+        options: { managed: false },
+      });
+      // Create a blocking span to control the transaction time and prevent it from closing automatically with partial batch responses.
+      // The blocking span needs to be ended manually when the request finishes.
+      const span = transaction?.startSpan(spanName, 'http-request', {
+        blocking: true,
+      });
```

Comment:
I wonder if we still need to use blocking spans for `managed: false` transactions. Couldn't we just close the transaction manually using `transaction?.end();`?

Reply:
Hi @xcrzx Yes you're right, it would be the same, the blocking span is only needed when `managed` is true. 
I had to switch `managed` to false in order to get accurate times, but with this option disabled we won't have the transaction traces in APM. 
The idea is that if we identify a high latency request and we want to investigate it, we can turn `managed` on here to see the transaction details in APM, without having to change any logic.

Reply:
Okay, I see, thanks 👍